File: 2399.cpp
class Solution {
public:
    bool checkDistances(string s, vector<int>& distance) {
        vector<int> first_index(26, -1);
        for (int i = 0; i < s.size(); i++) {
            int idx = s[i] - 'a';  
            if (first_index[idx] == -1) {
                first_index[idx] = i;  
            } else {
                if (i - first_index[idx] - 1 != distance[idx]) {
                    return false;
                }
            }
        }
        return true;
    }
};

File: 2399.py
class Solution:
    def checkDistances(self, s: str, distance: List[int]) -> bool:
        seen = {}
        for i, char in enumerate(s):
            idx = ord(char) - ord('a') 
            if char in seen:
                if i - seen[char] - 1 != distance[idx]:
                    return False
            else:
                seen[char] = i
        return True
